core/base/src/TROOT.cxx

```diff
@@ -577,6 +577,22 @@ namespace Internal {
 #endif
    }
 
+   ////////////////////////////////////////////////////////////////////////////////
+   /// Returns the maximum number of threads the user can instantiate.
```

Comment:
This comment is incorrect (at least for the !IMT case). Could you copy the comment from `GetMaxNThreadsAvailable()`?

Also this doesn't return the available threads but their number. I'd suggest to use the same name for both the `TROOT` member and the `ROOT::Internal` member.